lib/exercism/user_finished_tracks.rb

```diff
@@ -13,24 +13,27 @@ def initialize(user)
   def completed_tracks
     @completed_tracks ||= started_tracks.each_with_object([]) do |db_row, arr|
       track = tracks.find { |t| t.id == db_row['track_id'] }
-      arr << track if db_row['completed_problems_count'] >= track.problems.count
+      arr << track if completed?(db_row)
       arr
     end
   end
 
   private
 
   def started_tracks
-    ActiveRecord::Base.connection.execute(completed_count_sql).to_a.each do |db_row|
-      track = tracks.find { |t| t.id == db_row['track_id'] }
-      db_row['completed_problems_count'] = count_finished_problems(db_row['completed_problems'].split(','), track)
-    end
+    ActiveRecord::Base.connection.execute(completed_count_sql).to_a
   end
 
   def count_finished_problems(user_finished_exercises, track)
```

Comment:
@bernardoamc  
I also added this function because rubocop keeps giving me this error of Assignment Branch Condition size too high.